examples/tutorials/1_generating_signals.ipynb

```diff
@@ -0,0 +1,359 @@
+    "$$\n",
+    "\\langle \\rho^2 \\rangle_{\\vec{n}, \\psi} = \n",
+    "\\frac{1}{20}\n",
+    "\\frac{h_0^2 \\left(\\cos^4\\iota + 6 \\cos^2\\iota + 1\\right)}\n",
+    "{S_{\\textrm{n}}}\n",
+    "T_{\\textrm{obs}},\n",
+    "$$\n",
+    "where $T_{\\textrm{obs}}$ represents full duration of the data stream.\n",
+    "\n",
+    "The optimal SNR for a specific template can be computed using `pyfstat.SignalToNoiseRatio` assuming Gaussian noise with a given\n",
+    "`sqrtSX` value or from a specific set of SFTs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "57874a4b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "snr = pyfstat.SignalToNoiseRatio.from_sfts(\n",
+    "    F0=writer.F0, sftfilepath=writer.sftfilepath\n",
+    ")\n",
+    "squared_snr = snr.compute_snr2(\n",
+    "    Alpha=writer.Alpha, \n",
+    "    Delta=writer.Delta,\n",
+    "    psi=writer.psi,\n",
+    "    phi=writer.phi, \n",
+    "    h0=writer.h0,\n",
+    "    cosi=writer.cosi\n",
+    ")\n",
+    "print(f\"SNR: {np.sqrt(squared_snr)}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4970b4d9",
+   "metadata": {},
+   "source": [
+    "## Generating an ensemble of signals<a name=\"ensemble\"></a>\n",
+    "\n",
+    "Characterizing a CW search or follow-up method usually involves analyzing its response to an ensemble of signals\n",
+    "from a population of interest.\n",
+    "For the case of all-sky searches, this population is usually consistent with a uniform distribution of sources\n",
+    "across the sky with isotropic orientation.\n",
+    "\n",
+    "The `pyfstat.InjectionParametersGenerator` provides methods to draw parameters \n",
+    "from generic distributions in a suitable format, ready to be fed into `pyfstat.Writer`.\n",
+    "Uniform sampling across the sky is baked in the child class `pyfstat.AllSkyInjectionParametersGenerator`,\n",
+    "and isotropic priors on amplitude parameters are available in \n",
+    "`pyfstat.injection_parameters.isotropic_amplitude_priors`.\n",
+    "\n",
+    "All-sky searches tend to perform injection campaigns at fixed values of `h0` (or, equivalently, $\\mathcal{D}$)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "85da4362",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate 10 signals with parameters drawn from a specific population\n",
+    "num_signals = 10\n",
+    "\n",
+    "writer_kwargs ={\n",
+    "                \"tstart\": 1238166018,\n",
+    "                \"duration\": 365 * 86400,  \n",
+    "                \"detectors\": \"H1\",        \n",
+    "                \"sqrtSX\": 1e-23,          \n",
+    "                \"Tsft\": 1800,             \n",
+    "                \"SFTWindowType\": \"tukey\", \n",
+    "                \"SFTWindowBeta\": 0.01,\n",
+    "               }\n",
+    "\n",
+    "signal_parameters = pyfstat.AllSkyInjectionParametersGenerator(\n",
```

Comment:
Can you rename this variable `signal_parameters_generator` or something, to make it easier to run cells in different order, and make it more explicit that this is something different than the simple dict before.